pkg/security/module/grpc.go

```diff
@@ -26,13 +31,70 @@ type GRPCServer struct {
 	socketPath  string
 }
 
+type info struct {
+	credentials.CommonAuthInfo
+}
+
+// AuthType returns the type of info as a string.
+func (info) AuthType() string {
+	return "unix socket"
+}
+
+type grpcUnixSocketTransportCredential struct {
+	allowedUsrID int
+	allowedGrpID int
+}
+
+func (gustc grpcUnixSocketTransportCredential) ClientHandshake(ctx context.Context, authority string, conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	return conn, info{credentials.CommonAuthInfo{SecurityLevel: credentials.NoSecurity}}, nil
+}
+
+func (gustc grpcUnixSocketTransportCredential) ServerHandshake(conn net.Conn) (net.Conn, credentials.AuthInfo, error) {
+	unixConn, ok := conn.(*net.UnixConn)
+	if !ok {
+		return conn, info{credentials.CommonAuthInfo{SecurityLevel: credentials.NoSecurity}}, nil
+	}
+	valid, err := processnet.IsUnixNetConnValid(unixConn, gustc.allowedUsrID, gustc.allowedGrpID)
+	if err != nil || !valid {
+		if err != nil {
+			log.Errorf("unix socket %s -> %s closing connection, error %s", unixConn.LocalAddr(), unixConn.RemoteAddr(), err)
+		} else if !valid {
+			log.Debugf("unix socket %s -> %s closing connection, rejected", unixConn.LocalAddr(), unixConn.RemoteAddr())
```

Comment:
Do you think this log would be clear enough if seen out of context ?
In particular for the 2nd one I think explaining that 'the user is not allowed' would help.